### DISCUSSION TITLE
How to bind the public folder in docker?

### DISCUSSION THREAD

**Discussion by irlamplex** [Lissy93/dashy#314] [Help and Support] [answered]
Hi Lissy

I'm new to docker and wondered if you could help me to change the favicons and a few more things...

After i have installed dashy i can not see anything in my dashy folder, i.e a conf.yml file?
all i can see is a FOLDER named conf.yml not a real conf.yml file but a folder?

here is a snippet of my docker setup file:

`- name: Create dashy directories
  file: "path={{ item }} state=directory mode=0775 owner={{ user.name }} group={{ user.name }}"
  with_items:
    - /opt/dashy

- name: Create and start container
  docker_container:
    name: dashy
    image: lissy93/dashy
    pull: yes
    env:
      PUID: "{{ uid }}"
      PGID: "{{ gid }}"
      TZ: "{{ tz }}"
      VIRTUAL_HOST: "home.{{ user.domain }}"
      VIRTUAL_PORT: "8080"
      LETSENCRYPT_HOST: "home.{{ user.domain }}"
      LETSENCRYPT_EMAIL: "{{ user.email }}"
    volumes:
      - "/opt/dashy:/public/conf.yml"
    labels:
      "com.github.cloudbox.cloudbox_managed": "true"
    networks:
      - name: cloudbox
        aliases:
          - dashy
    purge_networks: yes
    restart_policy: unless-stopped
    state: started`

where am i going wrong, i just want to bind the whole public folder so i can edit any files etc...
i can use the built in config editor etc, everything is working. I'm accessing the dashboard through a reverse proxy (home.mydomain.com) i just cant manually edit anything in my  "opt/dashy" folder
as all i have in there is a another FOLDER called conf.yml

Any help would be much apricated.

Thanks

**Comment by lissy93** [comment, marked as answer]
So it's expecting a file for the config, but when you create a volume, it is a directory.

In my setup I do something like this (although there may be a better method)
```yaml
volumes:
  - /path/to/my-config.yml:/app/public/conf.yml
```

@EVOTk Is much better with Docker than me, so he may have some thoughts?

I think if Dashy were to accept a directory where all assets (config file, image icons, stylesheets, etc) could be passed into, that would be much better. Am considering changing this, but it would be a breaking change, so I would need to think through the best way to implement this.

**Comment by EVOTk** [comment]
Hello,
When creating a container, you need to provide the path to the conf.yml file as provided in the Dashy image.

Docker will not automatically create a file, you have to do it manually.

In /opt/dashy create an empty conf.yml file ( for exemple with `touch conf.yml` ), and modify your volume like this to be in accordance with Dashy:
```
volumes:
- "/opt/dashy/conf.yml:/public/conf.yml"
```

How did you create the Docker container?
